src/app/qgsattributetabledialog.cpp

```diff
@@ -351,6 +353,26 @@ void QgsAttributeTableDialog::updateFieldFromExpressionSelected()
   runFieldCalculation( mLayer, mFieldCombo->currentText(), mUpdateExpressionText->currentField(), filteredIds );
 }
 
+void QgsAttributeTableDialog::viewWillShowContextMenu( QMenu* menu, QModelIndex atIndex )
+{
+  if ( menu )
+  {
+    menu->addAction( tr( "Zoom to feature" ), this, SLOT( zoomToFeature() ) );
```

Comment:
Would here be the place to add a check for __geometryless__ layers, so the context menu will not show the option _Zoom to feature_ when that will not be possible anyway?